Backend/controller/NewWorldScraper/NewWorldScraper.js

```diff
@@ -60,102 +58,64 @@ async function fetchAllItemsOneCategory(category){
 
   let allItemsConfig = {
     method: 'get',
-    maxBodyLength: Infinity,
+    maxBodyLength: 1024 * 1024 * 512,
     url: url,
-    headers: { 
-      'Cookie': '__cf_bm=mSagMqawJXFKGXyGlx3F.PjlC8JQXz.leHE1tODhuGM-1687867900-0-AYsGTN7Y0dvsC1RuU3cBYA79kKdgWyoz6+f6sOvtXQwq1LfNM+6vZpvUI1a+ZaiyoL25uJhHBp3b5HSu0SkUw6T8ZmnfVopd/yu+ep3F4W7L'
-    }
   };
 
-  try {
-    let response = await axios.request(allItemsConfig);
-    console.log("this request Passed");
-    console.log(url);
-    let data = response.data;
-    return data;
-  } catch (error){
-    console.error(`ERROR: ${error}`);
-    console.log("This request failed");
-    console.log(url);
-    throw error;
-  }
+  let response = await axios.request(allItemsConfig);
+  let data = response.data;
+  return data;
 }
 
-let failCounter = 0;
+/**
+ * @param {Array<String>} categories 
+ */
 async function fetchAllitems(categories){
-  const maxRetries = 5;
-  let retryDelay = 2000;
-
-  for (let retryCount = 0; retryCount < maxRetries; retryCount++){
-    try{
-      // we must wait untill all the prmises from the fetch of each category is resolved
-      await Promise.all(
-        //map the fetchAllItems function to each category
-        categories.map(async (category) => {
-          let itemsFromCategory = await fetchAllItemsOneCategory(category);
-          //save the fetched items into the products array
-          getAllItems(itemsFromCategory);
-        })
-      );
-      break;
-    }catch (error){
-      console.error(`Error occurred: ${error}`);
```

Comment:
Catching here also means that if any requests fail, all requests will be cancelled and will have to start again